### ISSUE TITLE
how to auto mount dev drive when windows startup

### ISSUE THREAD

**Issue by Ddoudou** [microsoft/WinDev#121]
### Windows Build Number

10.0.22631.0

### Processor Architecture

AMD64

### Memory

16G

### Storage Type, free / capacity

SSD 1T

### Relevant apps installed

~

### Traces collected via Feedback Hub

~

### Isssue description

how to auto mount dev drive when windows startup

### Steps to reproduce

~

### Expected Behavior

auto mount dev drive when windows startup

### Actual Behavior

~

**Comment by adrastogi** [comment]
@Ddoudou can you elaborate on how you created your Dev Drive? Currently, I believe that only Dev Drives created via the Settings app will be automatically mounted. See the comment below (the broader thread may be of interest as well).

https://github.com/microsoft/devhome/issues/1903#issuecomment-2006184670

cc: @AdamBraden

**Comment by Ddoudou** [comment]
> @Ddoudou can you elaborate on how you created your Dev Drive? Currently, I believe that only Dev Drives created via the Settings app will be automatically mounted. See the comment below (the broader thread may be of interest as well).
> 
> [microsoft/devhome#1903 (comment)](https://github.com/microsoft/devhome/issues/1903#issuecomment-2006184670)
> 
> cc: @AdamBraden

use the devhome

**Comment by adrastogi** [comment]
@Ddoudou thanks! Did you create a physical drive or a virtual drive? Could you also please file a Feedback Hub item for this? You can invoke Feedback Hub via [Windows]+[f] and paste the link here so we can more easily find your information on the back end. The link can be found at the bottom of the feedback report. Please use "Developer Platform -> Dev Drive" for the Category.

**Comment by Ddoudou** [comment]
@adrastogi https://aka.ms/AAr0sxs

**Comment by adrastogi** [comment]
@Ddoudou sorry for the delayed response- appreciate you filing the feedback. Would you mind running the command below from a command prompt and sharing the output? 

`reg query HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\AutoAttachVirtualDisks /s`

**Comment by Ddoudou** [comment]
![image](https://github.com/user-attachments/assets/2caf4802-f509-4721-9dd7-4a385ec085f5)
@adrastogi no output

**Comment by Eli-Black-Work** [comment]
@Ddoudou BTW, you might want to post this at the DevHome repo: https://github.com/microsoft/devhome :)

**Comment by rangasurendra** [comment]
There is a workaround, you can go to `Settings > System > For developers` at the bottom of the screen under the `Dev Drive` go to `Create a Dev Drive`. And then on the next screen click `Attach VHD` - attach your existing dev drive VHDX

In my case had to move virtual dev drive [created through `Dev Home` few months back] to a better performing SSD. Which broke the auto mount at startup. Finally, this worked for me.

**Comment by joncamp** [comment]
As noted in https://github.com/microsoft/devhome/issues/1903#issuecomment-2610467676, seeing this on my arm64 surface. Dev drive was created via the dev home interface. Virtual disks service is set to auto start. I can manually attach the drive fine after boot using the diskpart or GUI approaches noted in the thread.

```
❯ reg query HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\AutoAttachVirtualDisks /s

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\AutoAttachVirtualDisks\{ce8d65e8-1427-44f2-9c8f-43b3cdcb9f1f}
    Path    REG_SZ    C:\DevDrive.vhdx
```

**Comment by GF-Huang** [comment]
> Open Windows Settings, go to "Disks & volumes", click on "Attach VHD", and find your VHD or VHDX file. It will stay mounted across reboots.
>
> If you instead mount the virtual disk by double-clicking it, it won't auto-mount after a reboot.

https://superuser.com/a/1882252/986246

**Comment by niikoo** [comment]
This is how I set it up.

### Task Scheduler

**General**  
Name: Auto-Mount DevDrive VHDX at Startup  
When running the task, use the following user account: \[I selected an account with administrative rights\]  
Selected `Run whether user is logged in or not` , `Do not store password` and `Run with highest privileges`.

**Triggers**

*   At startup
*   At log on

**Actions**  
Action: `Start a program`  
Program/script: `diskpart.exe`  
Arguments: `/S "C:\devdrive.automount.diskpart"`

**Conditions**  
Unchecked everything.

**Settings**  
Checked the top 5 checkboxes.  
Set `Do not start a new instance` on the last dropdown.

### Create the diskpart file

Path: `"C:\devdrive.automount.diskpart"`  
File contents:

    select vdisk file="C:\devDrive.vhdx"
    attach vdisk
    exit
    

Replace `"C:\devDrive.vhdx"` with the path to your DevDrive VHDX file.

**Comment by neominky** [comment]
> > Open Windows Settings, go to "Disks & volumes", click on "Attach VHD", and find your VHD or VHDX file. It will stay mounted across reboots.
> > If you instead mount the virtual disk by double-clicking it, it won't auto-mount after a reboot.
> 
> https://superuser.com/a/1882252/986246

No, the mount status is not maintained after rebooting. So I searched and found this place. System is Windows Server 2025
